csrc/host_ir/jit.cpp

```diff
@@ -68,185 +77,281 @@ T throwIfError(llvm::Expected<T>&& E) {
+  llvm::Function* allocate_tensor_func =
+      mod->getFunction(kAllocateTensorFuncName);
+  llvm::Value* out_tensor_ptr =
+      builder.CreateCall(allocate_tensor_func, {}, "out_tensor_raw");
+
+  // set the output tensor to the input tensor
+  llvm::Function* set_tensor_func = mod->getFunction(kSetTensorFuncName);
+  builder.CreateCall(set_tensor_func, {out_tensor_ptr, in_tensor_ptr});
+
+  // bind the output tensor to val_to_value
+  val_to_value[out_tv] = out_tensor_ptr;
+
+  if (isDebugDumpEnabled(DebugDumpOption::HostIrJit)) {
+    auto* func = builder.GetInsertBlock()->getParent();
+    llvm::outs() << "=== LLVM IR Generation for LoadStoreOp ===\n";
+    func->print(llvm::outs(), nullptr);
+  }
+}
 
-  const bool debug_print = isDebugDumpEnabled(DebugDumpOption::HostIrJit);
-  if (debug_print) {
-    llvm::outs() << "=== LLVM IR ===\n";
-    mod->print(llvm::outs(), nullptr);
+void compileMainFuncInputs(
+    const hir::HostIrContainer* container,
+    llvm::IRBuilder<>& builder,
+    std::unordered_map<Val*, llvm::Value*>& val_to_value) {
+  llvm::LLVMContext& ctx = builder.getContext();
+
+  // Get the current function (main) and its first argument
+  llvm::Function* func = builder.GetInsertBlock()->getParent();
+  llvm::Value* aten_tensor_array_ptr = func->getArg(0);
+
+  llvm::Type* aten_tensor_array_type = getInt8PtrDynamicArrayType(ctx);
+  // bind input aten tensor sizes to val_to_value
+  for (size_t i = 0; i < container->inputs().size(); ++i) {
+    auto* input = container->inputs()[i];
+    if (TensorView* tv = dynamic_cast<TensorView*>(input)) {
+      llvm::Value* aten_tensor_ptr_addr = builder.CreateGEP(
+          aten_tensor_array_type, aten_tensor_array_ptr, {builder.getInt64(i)});
+      aten_tensor_ptr_addr->setName("input_aten_tensor_addr");
+      // Load the actual tensor pointer from the array
+      llvm::Value* aten_tensor_ptr =
+          builder.CreateLoad(getInt8PtrType(ctx), aten_tensor_ptr_addr);
+      aten_tensor_ptr->setName("input_aten_tensor");
+      // bind input aten tensor sizes to val_to_value
+      auto logical_domain = TensorDomain::noReductions(tv->getLogicalDomain());
+      // TODO: We should validate const size and strides here, ie. dim check
+      for (size_t dim = 0; dim < logical_domain.size(); ++dim) {
+        if (logical_domain[dim]->isBroadcast()) {
+          val_to_value[logical_domain[dim]->extent()] = builder.getInt64(1);
+          if (logical_domain[dim]->hasExpandedExtent()) {
+            val_to_value[logical_domain[dim]->expandedExtent()] =
+                generateTensorSizeExtraction(aten_tensor_ptr, dim, builder);
+          }
+        } else {
+          val_to_value[logical_domain[dim]->extent()] =
+              generateTensorSizeExtraction(aten_tensor_ptr, dim, builder);
+        }
+      }
+      // bind input aten tensor to val_to_value
+      val_to_value[tv->as<Val>()] = aten_tensor_ptr;
+    } else {
+      NVF_THROW("Unsupported expression type: ", input);
```

Comment:
Nit: Instead of 
```
if (auto* tv = dynamic_cast<TensorView*>(...)) {
  A
} else {
  NVF_THROW(...);
}
```
, do 
```
auto* tv = dynamic_cast<TensorView*>(...);
NVF_ERROR(tv != nullptr, ...);
A
```
for less indentation.